src/Nancy/NancyEngine.cs

```diff
@@ -252,7 +253,21 @@ private void CheckStatusCodeHandler(NancyContext context)
                 return;
             }
 
-            handler.Handle(context.Response.StatusCode, context);
+            try
+            {
+                handler.Handle(context.Response.StatusCode, context);
+            }
+            catch (Exception ex)
+            {
+                if (defaultHandler != null)
+                {
+                    defaultHandler.Handle(context.Response.StatusCode, context);
+                }
+                else
+                {
+                    throw new Exception("No default status code handler found", ex);

```

Comment:
Not really. I just mangles the stack trace. The fact that a status code handler wasn't found isn't the source of the exception. That's fine.